### PR TITLE
Adding conv1d and conv3d

### PR DESCRIPTION
**Pull request by hafezgh**
The conv1d and conv3d classes and their corresponding learning rule methods are added.

### CODE REVIEW

File: test/network/test_conv1d.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
+### Toy example to test Conv1dConnection (the dataset used is MNIST but each image is raveled (each sample has shape (784,)).
+
```
Comment (Hananel-Hazan) [inline test/network/test_conv1d.py:2]: This code should be at MNIST examples. 
Test directory contain tests scripts that perform a integrity check for the models.

File: test/network/test_conv3d.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
+### Toy example to test Conv3dConnection (the dataset used is MNIST but with a dimension replicated 
+### for each image (each sample has size (28, 28, 28))
```
Comment (Hananel-Hazan) [inline test/network/test_conv3d.py:2]: This code should be at MNIST examples. 
Test directory contain tests scripts that perform a integrity check for the models.

**Review by Hananel-Hazan — CHANGES_REQUESTED** [review]

**Comment by hafezgh** [comment]
I relocated the files.

**Comment by Hananel-Hazan** [comment]
Thank you Hafez for the great work !

**Review by Hananel-Hazan — APPROVED** [review]

**Pull request merged**